src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs

```diff
@@ -1301,9 +1301,9 @@ internal unsafe void AdjustBytes(int count)
             internal unsafe byte GetNextByte()
             {
                 Debug.Assert(_bytes < _byteEnd, "[EncodingCharBuffer.GetNextByte]Expected more date");
-                if (_bytes >= _byteEnd)
-                    return 0;
-                return *(_bytes++);
+                if (_bytes < _byteEnd)
```

Comment:
I do not think that this is more readable with this change. The original code used early exit for exceptional condition that is generally easier to understand.  I really like how LLVM coding guidelines describe it: https://llvm.org/docs/CodingStandards.html#use-early-exits-and-continue-to-simplify-code . This one is too small of a method to clearly see the benefits of the early exit pattern, but it is nice to be consistent.

Reply:
However, when you refactor it, and see the if statement and the asserts are the same, this raises the question: does the assert need to be there?

Reply:
This code operates on unsafe pointers. It is not unusual to both assert and check in code like this  prone to buffer overruns for defense in depth.

Reply:
Got it.